packages/shared/src/withContext/withContext.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+
+import * as React from 'react';
+import isEqual from 'react-fast-compare';
+
+export default function withContext<T>(
+  select: T => Object,
+  Consumer: React.ComponentType<any> | React.StatelessFunctionalComponent<any>,
+) {
+  return (Component: React.ElementType) => {
+    class WithShouldComponentUpdate extends React.Component<{|
+      ...T,
+      +children: React.Element<any>,
+    |}> {
+      // $FlowExpectedError: We need to pass on the navigationOptions if any, flow does not know about it, but a react component might have it
+      static navigationOptions = Component.navigationOptions;
+
+      /**
+       * All components that subscribes to context would re render even when props they are not subsrcibing
+       * too changes. This shouldComponentUpdate should make the subscriber only re render when its
+       * props are changing
+       */
+      shouldComponentUpdate(nextProps: T) {
+        return !isEqual(select(nextProps), select(this.props));
+      }
+
+      render() {
+        const children = React.cloneElement(this.props.children, {
+          ...(this.props: Object),
+          testID: 'withContext',
```

Comment:
Will this overwrite the `testID` we might be using for e2e testing for instance?

Reply:
You are right! I should do this differently